Source/WebCore/platform/animation/AcceleratedTimeline.h

```diff
@@ -36,19 +37,29 @@ namespace WebCore {
 class AcceleratedTimeline : public RefCounted<AcceleratedTimeline> {
     WTF_MAKE_TZONE_ALLOCATED(AcceleratedTimeline);
 public:
-    WEBCORE_EXPORT static Ref<AcceleratedTimeline> create(const TimelineIdentifier&, Seconds originTime);
+
+    static Ref<AcceleratedTimeline> create(const TimelineIdentifier&, Seconds originTime);
+    static Ref<AcceleratedTimeline> create(const TimelineIdentifier&, ProgressResolutionData);
+
+    bool isMonotonic() const { return !!m_originTime; }
+    bool isProgressBased() const { return !isMonotonic(); }
 
     // Encoding support.
+    WEBCORE_EXPORT static Ref<AcceleratedTimeline> create(TimelineIdentifier&&, std::optional<Seconds>&& originTime, std::optional<ProgressResolutionData>&&);
     const TimelineIdentifier& identifier() const { return m_identifier; }
-    const Seconds originTime() const { return m_originTime; }
+    const std::optional<Seconds> originTime() const { return m_originTime; }
+    const std::optional<ProgressResolutionData> progressResolutionData() const { return m_progressResolutionData; }
 
-    WEBCORE_EXPORT virtual ~AcceleratedTimeline() = default;
+    virtual ~AcceleratedTimeline() = default;
 
 private:
     AcceleratedTimeline(const TimelineIdentifier&, Seconds originTime);
+    AcceleratedTimeline(const TimelineIdentifier&, ProgressResolutionData);
+    AcceleratedTimeline(TimelineIdentifier&&, std::optional<Seconds>&& originTime, std::optional<ProgressResolutionData>&&);
 
     TimelineIdentifier m_identifier;
-    Seconds m_originTime;
+    std::optional<Seconds> m_originTime;
+    std::optional<ProgressResolutionData> m_progressResolutionData;
```

Comment:
Variant, so that only one of these two can be valid?

Reply:
This was addressed in the latest patch.